gcloud/bigtable/happybase/table.py

```diff
@@ -127,6 +130,88 @@ def regions(self):
         raise NotImplementedError('The Cloud Bigtable API does not have a '
                                   'concept of splitting a table into regions.')
 
+    def counter_get(self, row, column):
+        """Retrieve the current value of a counter column.
+
+        This method retrieves the current value of a counter column. If the
+        counter column does not exist, this function initializes it to ``0``.
+
+        .. note::
+
+            Application code should **never** store a counter value directly;
+            use the atomic :meth:`counter_inc` and :meth:`counter_dec` methods
+            for that.
+
+        :type row: str
+        :param row: Row key for the row we are getting a counter from.
+
+        :type column: str
+        :param column: Column we are ``get``-ing from; of the form ``fam:col``.
+
+        :rtype: int
+        :returns: Counter value (after initializing / incrementing by 0).
+        """
+        # Don't query directly, but increment with value=0 so that the counter
+        # is correctly initialized if didn't exist yet.
+        return self.counter_inc(row, column, value=0)
+
+    def counter_inc(self, row, column, value=1):
+        """Atomically increment a counter column.
+
+        This method atomically increments a counter column in ``row``.
+        If the counter column does not exist, it is automatically initialized
+        to ``0`` before being incremented.
+
+        :type row: str
+        :param row: Row key for the row we are incrementing a counter in.
+
+        :type column: str
+        :param column: Column we are incrementing a value in; of the
+                       form ``fam:col``.
+
+        :type value: int
+        :param value: Amount to increment the counter by. (If negative,
+                      this is equivalent to decrement.)
+
+        :rtype: int
+        :returns: Counter value after incrementing.
+        """
+        row = self._low_level_table.row(row)
+        if isinstance(column, six.binary_type):
+            column = column.decode('utf-8')
```

Comment:
The `column` variable is passed by the user so I was allowing for them to do as they please. The HappyBase concept of `fam:` or `fam:col` is different from that of Cloud Bigtable.

In Cloud Bigtable the fields are separate: [`Column.qualifier`][1] (equivalent of `col` in `fam:col`) is `bytes` and [`Family.name`][2] (i.e. `fam`) is a string.

It's unclear what the right thing here is. For example if `fam` is a string but `col` has non-ascii bytes, then decoding to `utf-8` may be a bad idea.

[1]: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/7c77c170c8818a0440e0d2f4ecedcb8329c1f8d9/bigtable-protos/src/main/proto/google/bigtable/v1/bigtable_data.proto#L58
[2]: https://github.com/GoogleCloudPlatform/cloud-bigtable-client/blob/7c77c170c8818a0440e0d2f4ecedcb8329c1f8d9/bigtable-protos/src/main/proto/google/bigtable/v1/bigtable_data.proto#L45